src/fsharp/InfoReader.fs

```diff
@@ -700,7 +702,34 @@ let TryFindIntrinsicMethInfo infoReader m ad nm ty =
 /// Try to find a particular named property on a type. Only used to ensure that local 'let' definitions and property names
 /// are distinct, a somewhat adhoc check in tc.fs.
 let TryFindPropInfo infoReader m ad nm ty = 
-    GetIntrinsicPropInfosOfType infoReader (Some nm) ad AllowMultiIntfInstantiations.Yes IgnoreOverrides m ty 
+    GetIntrinsicPropInfosOfType infoReader (Some nm) ad AllowMultiIntfInstantiations.Yes IgnoreOverrides m ty
+    
+/// Try to find the method that overrides the given method from the given type.
+/// The given type must be an IL type.
+/// The given method must be an IL method.
+let TryFindILIntrinisicOverriderMethInfo (infoReader: InfoReader) m ad ty (overrides: MethInfo) =
+    let g = infoReader.g
+
+    match tryAppTy g ty, overrides with
+    | ValueSome (tcref, _), ILMeth (_, ilMethInfo, _) when tcref.IsILTycon ->
+        let ilMeth = ilMethInfo.ILMethodRef
+
+        tcref.ILTyconRawMetadata.MethodImpls.AsList
```

Comment:
I need to check the `overrides: MethInfo` enclosing type with the enclosing type found in `ilMethImpl.Overrides` to ensure we are indeed overriding correctly.